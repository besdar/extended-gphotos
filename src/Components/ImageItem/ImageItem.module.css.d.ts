export const __esModule: true;
export const image: string;
export const imageEnter: string;
export const imageEnterActive: string;
export const imageEnterDone: string;
export const imageExit: string;
export const imageExitActive: string;
export const imageExitDone: string;

