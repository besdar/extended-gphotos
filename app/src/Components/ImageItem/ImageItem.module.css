.image {
    width: 100%;
    height: 100%;
    background-repeat: no-repeat;
    background-size: cover;
    background-position: center;
    cursor: pointer;
}

/* ENTER TRANSITION */
/* Declare transition start properties*/
.imageEnter {
    opacity: 0;
}

/* Declare transition properties */
.imageEnterActive {
    opacity: 1;
    transition-property: all;
    transition-duration: 2s;
    transition-timing-function: linear;
}

.imageEnterDone {
    opacity: 1;
}

/* EXIT TRANSITION */
.imageExit {
    opacity: 1;
}

.imageExitActive {
    opacity: 0;
    transition-property: all;
    transition-duration: 2s;
    transition-timing-function: linear;
}

.imageExitDone {
    opacity: 0;
}