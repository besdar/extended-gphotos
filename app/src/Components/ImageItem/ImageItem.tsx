import React, { FC, useRef } from 'react';
import { CSSTransition } from 'react-transition-group';
import s from './ImageItem.module.css';
import { SquareType } from '../GalleryStructure';

export type ImageItemType = {
    imageUrl: string,
    tooltipTitle?: string,
    webUrl?: string,
}

export type ImageItemPropsType = ImageItemType & {
    anotherImageUrl: string,
    isFirstImageActive: boolean,
    squareCoordinates?: SquareType
}

const transitionClassNames = {
  enter: s.imageEnter,
  exit: s.imageExit,
  enterActive: s.imageEnterActive,
  enterDone: s.imageEnterDone,
  exitDone: s.imageExitDone,
  exitActive: s.imageExitActive,
};

const ImageItem: FC<ImageItemPropsType> = ({
  isFirstImageActive,
  imageUrl,
  anotherImageUrl,
  squareCoordinates,
  tooltipTitle,
  webUrl,
}) => {
  const firstImage = useRef<HTMLImageElement>(null);
  const secondImage = useRef<HTMLImageElement>(null);
  const gridArea = squareCoordinates && `${squareCoordinates.ystart}/${squareCoordinates.xstart}/${squareCoordinates.yend}/${squareCoordinates.xend}`;

  return (
    <>
      <CSSTransition
        in={isFirstImageActive}
        timeout={3000}
        mountOnEnter
        unmountOnExit
        classNames={transitionClassNames}
        nodeRef={firstImage}
      >
        <a href={webUrl} target="_blank" rel="noreferrer">
          <img
            title={tooltipTitle}
            alt={tooltipTitle}
            onClick={() => window.open(webUrl)}
            ref={firstImage}
            className={s.image}
            src={imageUrl}
            style={{ gridArea }}
            aria-hidden
          />
        </a>
      </CSSTransition>
      <CSSTransition
        in={!isFirstImageActive}
        timeout={3000}
        mountOnEnter
        unmountOnExit
        classNames={transitionClassNames}
        nodeRef={secondImage}
      >
        <a href={webUrl} target="_blank" rel="noreferrer">
          <img
            title={tooltipTitle}
            alt={tooltipTitle}
            ref={secondImage}
            className={s.image}
            src={anotherImageUrl}
            style={{ gridArea }}
          />
        </a>
      </CSSTransition>
    </>
  );
};

export default React.memo(ImageItem);
