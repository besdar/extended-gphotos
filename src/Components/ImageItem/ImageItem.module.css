.image {
  width: 100%;
  height: 100%;
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
  cursor: pointer;

  /* object-fit: cover; */
}

/* Declare transition start properties */
.imageEnter {
  opacity: 0;
}

/* Declare transition properties */
.imageEnterActive {
  opacity: 1;
  transition-property: all;
  transition-duration: 3s;
  transition-timing-function: linear;
}

.imageEnterDone {
  opacity: 1;
}

.imageExit {
  opacity: 1;
}

.imageExitActive {
  opacity: 0;
  transition-property: all;
  transition-duration: 3s;
  transition-timing-function: linear;
}

.imageExitDone {
  opacity: 0;
}
