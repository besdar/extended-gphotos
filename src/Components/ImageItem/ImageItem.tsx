import React, { FC, useRef } from 'react';
import { CSSTransition } from 'react-transition-group';
import * as s from './ImageItem.module.css';
import { SquareType } from '../GalleryStructure';

export type ImageItemType = {
  imageUrl: string,
  tooltipTitle?: string,
  webUrl?: string,
}

export type ImageItemPropsType = ImageItemType & {
  anotherImageUrl: string,
  isFirstImageActive: boolean,
  squareCoordinates?: SquareType
}

const transitionClassNames = {
  enter: s.imageEnter,
  exit: s.imageExit,
  enterActive: s.imageEnterActive,
  enterDone: s.imageEnterDone,
  exitDone: s.imageExitDone,
  exitActive: s.imageExitActive,
};

const ImageItem: FC<ImageItemPropsType> = ({
  isFirstImageActive,
  imageUrl,
  anotherImageUrl,
  squareCoordinates,
  tooltipTitle,
  webUrl,
}) => {
  const firstImage = useRef<HTMLAnchorElement>(null);
  const secondImage = useRef<HTMLAnchorElement>(null);
  const gridArea = squareCoordinates && `${squareCoordinates.ystart}/${squareCoordinates.xstart}/${squareCoordinates.yend}/${squareCoordinates.xend}`;

  // TODO: Change null inside A tag onto img
  return (
    <>
      <CSSTransition
        in={isFirstImageActive}
        timeout={3000}
        mountOnEnter
        unmountOnExit
        classNames={transitionClassNames}
        nodeRef={firstImage}
      >
        <a
          href={webUrl}
          target="_blank"
          rel="noreferrer"
          className={s.image}
          style={{ gridArea, backgroundImage: `url(${imageUrl})` }}
          title={tooltipTitle}
          ref={firstImage}
        >
          {null}
        </a>
      </CSSTransition>
      <CSSTransition
        in={!isFirstImageActive}
        timeout={3000}
        mountOnEnter
        unmountOnExit
        classNames={transitionClassNames}
        nodeRef={secondImage}
      >
        <a
          href={webUrl}
          target="_blank"
          rel="noreferrer"
          className={s.image}
          style={{ gridArea, backgroundImage: `url(${anotherImageUrl})` }}
          title={tooltipTitle}
          ref={secondImage}
        >
          {null}
        </a>
      </CSSTransition>
    </>
  );
};

export default React.memo(ImageItem);
