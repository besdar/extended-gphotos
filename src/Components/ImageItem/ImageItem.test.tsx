import React from 'react';
import { mount } from 'enzyme';
import ImageItem from './ImageItem';

jest.useFakeTimers();

it('Correct render', () => {
  const component = mount(<ImageItem
      anotherImageUrl="anotherImg"
      isFirstImageActive
      imageUrl="firstImg"
      tooltipTitle="tooltip"
      squareCoordinates={{
        ystart: 1,
        xend: 2,
        xstart: 3,
        yend: 4,
      }}
      webUrl="weburl"
    />);

  expect(component).toMatchSnapshot();

  component.setProps({ isFirstImageActive: false });

  expect(component).toMatchSnapshot();
});