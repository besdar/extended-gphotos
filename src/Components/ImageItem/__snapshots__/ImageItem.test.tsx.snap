// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correct render 1`] = `
<Fragment>
  <CSSTransition
    classNames={
      Object {
        "enter": "imageEnter",
        "enterActive": "imageEnterActive",
        "enterDone": "imageEnterDone",
        "exit": "imageExit",
        "exitActive": "imageExitActive",
        "exitDone": "imageExitDone",
      }
    }
    in={true}
    mountOnEnter={true}
    nodeRef={
      Object {
        "current": null,
      }
    }
    timeout={3000}
    unmountOnExit={true}
  >
    <a
      className="image"
      href="weburl"
      rel="noreferrer"
      style={
        Object {
          "backgroundImage": "url(firstImg)",
          "gridArea": "1/3/4/2",
        }
      }
      target="_blank"
      title="tooltip"
    />
  </CSSTransition>
  <CSSTransition
    classNames={
      Object {
        "enter": "imageEnter",
        "enterActive": "imageEnterActive",
        "enterDone": "imageEnterDone",
        "exit": "imageExit",
        "exitActive": "imageExitActive",
        "exitDone": "imageExitDone",
      }
    }
    in={false}
    mountOnEnter={true}
    nodeRef={
      Object {
        "current": null,
      }
    }
    timeout={3000}
    unmountOnExit={true}
  >
    <a
      className="image"
      href="weburl"
      rel="noreferrer"
      style={
        Object {
          "backgroundImage": "url(anotherImg)",
          "gridArea": "1/3/4/2",
        }
      }
      target="_blank"
      title="tooltip"
    />
  </CSSTransition>
</Fragment>
`;

exports[`Correct render 2`] = `
<Fragment>
  <CSSTransition
    classNames={
      Object {
        "enter": "imageEnter",
        "enterActive": "imageEnterActive",
        "enterDone": "imageEnterDone",
        "exit": "imageExit",
        "exitActive": "imageExitActive",
        "exitDone": "imageExitDone",
      }
    }
    in={false}
    mountOnEnter={true}
    nodeRef={
      Object {
        "current": null,
      }
    }
    timeout={3000}
    unmountOnExit={true}
  >
    <a
      className="image"
      href="weburl"
      rel="noreferrer"
      style={
        Object {
          "backgroundImage": "url(firstImg)",
          "gridArea": "1/3/4/2",
        }
      }
      target="_blank"
      title="tooltip"
    />
  </CSSTransition>
  <CSSTransition
    classNames={
      Object {
        "enter": "imageEnter",
        "enterActive": "imageEnterActive",
        "enterDone": "imageEnterDone",
        "exit": "imageExit",
        "exitActive": "imageExitActive",
        "exitDone": "imageExitDone",
      }
    }
    in={true}
    mountOnEnter={true}
    nodeRef={
      Object {
        "current": null,
      }
    }
    timeout={3000}
    unmountOnExit={true}
  >
    <a
      className="image"
      href="weburl"
      rel="noreferrer"
      style={
        Object {
          "backgroundImage": "url(anotherImg)",
          "gridArea": "1/3/4/2",
        }
      }
      target="_blank"
      title="tooltip"
    />
  </CSSTransition>
</Fragment>
`;
